### DISCUSSION TITLE
Network partition between 2 nodes in HA setup causes reachable node's API not to respond

### DISCUSSION THREAD

**Discussion by renannprado** [k3s-io/k3s#6154] [Q&A] [answered]
I have executed [this setup](https://rancher.com/docs/k3s/latest/en/installation/ha-embedded/), so basically I have this:

```
NAME       STATUS   ROLES                       AGE     VERSION
node1   Ready    control-plane,etcd,master   2d13h   v1.24.4+k3s1
node2   Ready    control-plane,etcd,master   15h     v1.24.4+k3s1
```

`node1` is in my home network, while `node2` is in the cloud, they are `masters` as well as `workers` 😄 . Both nodes are connected through wireguard.
I wanted to test what happens if there's a network partition, so I switched off the wireguard connection between the two.

After doing that, when I run `kubectl get pods` I get:

```
Get "https://192.168.169.130:6443/api/v1/namespaces/default/pods?limit=500": dial tcp 192.168.169.130:6443: connect: connection refused - error from a previous attempt: unexpected EOF
```

Even though the port on the `node1` is still reachable:

```
# nc -vz 192.168.169.130 6443
Connection to 192.168.169.130 port 6443 [tcp/sun-sr-https] succeeded!
```

I know I should be setting up 3 nodes minimum, but I would like to stick with 1 powerful node at home instead of two smaller ones, that's a conscious choice. If that's a cause of this issue, then I would consider having a 3 node setup, otherwise I would like to resolve it so that this doesn't happen when, for example, I don't have internet at home, therefore killing the connection between the two nodes.

**Comment by brandond** [comment, marked as answer]
Duplicate of https://github.com/k3s-io/k3s/issues/5210 - see https://github.com/k3s-io/k3s/issues/5210#issuecomment-1059073954

tl;dr you need an odd number of etcd nodes for quorum. Two nodes gets you pretty much nothing.
